### PR TITLE
Documentation for Kaltura <= 13.1.0 RCE (CVE-2017-14143)

### PR DESCRIPTION
**Pull request by Pushpamk**
Documentation for Kaltura <= 13.1.0 RCE (CVE-2017-14143) , PR #9114

### CODE REVIEW

**Comment by bwatters-r7** [comment]
@Pushpamk please add this file to your original PR.  All you need to do is check out the branch you originally pushed from your remote repo, add this file to it, commit it, and push it back up.  Your PR will be modified when you modify your branch in your remote repo.

**Comment by Pushpamk** [comment]
@bwatters-r7 but the original PR was not mine.   The original PR was  #9114 .

**Comment by bwatters-r7** [comment]
Oh; well that's different, then.  My apoligies that I assumed the PR was yours.  If @rverton has his repo set to allow pushes to his brnach, you could add his remote repo to your local machine, checkout that branch from him, add the file, and then make a PR to @rverton directly.  If @rverton accepts it, then they will land your PR into their branch and it will update the PR.  That would be my preference, though if that's not possible, I think this could remain as a stand-alone dependent PR, and I can sort it out.

File: documentation/modules/exploit/linux/http/kaltura_unserialize_cookie_rce.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+ 1. Start msfconsole
+ 2. use exploit/linux/http/kaltura_unserialize_cookie_rce
+ 3. set RHOST https://example.com (or IP address)
+ 4. set ENTRYID 0_xxxxxxxx
+ 5. set payload generic/custom
+ 6. set payloadstr "system('command you want to execute, eg.- ls -la');"
+ 7. run
+
+
+## Options
```
Comment (h00die) [inline documentation/modules/exploit/linux/http/kaltura_unserialize_cookie_rce.md]: i dont think this section is needed

File: documentation/modules/exploit/linux/http/kaltura_unserialize_cookie_rce.md [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+## Vulnerable Application
+
+ This module exploits a remote code execution within the Kaltura(<=13.1.0) via a cookie deserialization.
+ Vulnerability reference- https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2017-14143.
+
+
+## Verification Steps
+
+ 1. Start msfconsole
+ 2. use exploit/linux/http/kaltura_unserialize_cookie_rce
```
Comment (h00die) [inline documentation/modules/exploit/linux/http/kaltura_unserialize_cookie_rce.md]: wrap in ticks (`)

File: documentation/modules/exploit/linux/http/kaltura_unserialize_cookie_rce.md [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+ 3. set RHOST https://example.com (or IP address)
```
Comment (h00die) [inline documentation/modules/exploit/linux/http/kaltura_unserialize_cookie_rce.md]: wrap in ticks (`)

File: documentation/modules/exploit/linux/http/kaltura_unserialize_cookie_rce.md [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+ 4. set ENTRYID 0_xxxxxxxx
```
Comment (h00die) [inline documentation/modules/exploit/linux/http/kaltura_unserialize_cookie_rce.md]: wrap in ticks (`)

File: documentation/modules/exploit/linux/http/kaltura_unserialize_cookie_rce.md [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+ 5. set payload generic/custom
```
Comment (h00die) [inline documentation/modules/exploit/linux/http/kaltura_unserialize_cookie_rce.md]: wrap in ticks (`)

File: documentation/modules/exploit/linux/http/kaltura_unserialize_cookie_rce.md [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+ 6. set payloadstr "system('command you want to execute, eg.- ls -la');"
```
Comment (h00die) [inline documentation/modules/exploit/linux/http/kaltura_unserialize_cookie_rce.md]: wrap in ticks (`)

File: documentation/modules/exploit/linux/http/kaltura_unserialize_cookie_rce.md [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+ 7. run
```
Comment (h00die) [inline documentation/modules/exploit/linux/http/kaltura_unserialize_cookie_rce.md]: wrap in ticks (`)

File: documentation/modules/exploit/linux/http/kaltura_unserialize_cookie_rce.md (hunk 5 above)
Comment (h00die) [inline documentation/modules/exploit/linux/http/kaltura_unserialize_cookie_rce.md]: this isn't what you did in the example/scenario

File: documentation/modules/exploit/linux/http/kaltura_unserialize_cookie_rce.md (hunk 6 above)
Comment (h00die) [inline documentation/modules/exploit/linux/http/kaltura_unserialize_cookie_rce.md]: this isn't what you did in the example/scenario

File: documentation/modules/exploit/linux/http/kaltura_unserialize_cookie_rce.md [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+
+
+## Options
+
+ default RPORT 4444
+
+
+## Scenarios
+
```
Comment (h00die) [inline documentation/modules/exploit/linux/http/kaltura_unserialize_cookie_rce.md]: Typically we like a 3 # and then info about the scenario.  Is this 13.1? 12.0? I see its linux and most likely debian/ubuntu, its nice to know the actual system.
Example: ```### Kaltura 13.0 on Ubuntu 16.04 x64```

File: documentation/modules/exploit/linux/http/kaltura_unserialize_cookie_rce.md [hunk 9]
```diff
@@ -0,0 +1,46 @@
+## Description 
+
+The getUserzoneCookie function in Kaltura before 13.2.0 uses a hardcoded cookie secret to validate cookie signatures, which allows remote attackers to bypass an intended protection mechanism and consequently conduct PHP object injection attacks and execute arbitrary PHP code via a crafted userzone cookie. 
+ 
+
+## Vulnerable Application
+
+ This module exploits a remote code execution within the Kaltura(<=13.1.0) via a cookie deserialization.
+ Vulnerability reference- https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2017-14143.
```
Comment (h00die) [inline documentation/modules/exploit/linux/http/kaltura_unserialize_cookie_rce.md]: Is there no install instructions? or a .deb or something else you could help here to get someone to install?  Even just referencing the install docs from Kaltura is good start.

**Comment by bwatters-r7** [comment]
FYI, I feel bad I have not gotten back to this.  If there's any chance that you can send a pcap of it working, we can approve it without setting up Kaltura.

**Comment by bwatters-r7** [comment]
# Release Notes
This add documentation for `use exploit/linux/http/kaltura_unserialize_cookie_rce`

**Pull request merged**